st2common/st2common/models/db/auth.py

```diff
@@ -49,4 +53,32 @@ class TokenDB(stormbase.StormFoundationDB):
                             help_text='Arbitrary metadata associated with this token')
 
 
-MODELS = [UserDB, TokenDB]
+class ApiKeyDB(stormbase.StormFoundationDB, stormbase.UIDFieldMixin):
+    """
+    """
+    RESOURCE_TYPE = ResourceType.API_KEY
+    UID_FIELDS = ['id']
+
+    user = me.StringField(required=True)
```

Comment:
Not foreign key, it is username assuming name is unique. I could switch to ID but then it requires a join for the result of ApiKeyDB to be useful.